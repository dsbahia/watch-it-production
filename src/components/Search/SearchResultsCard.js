import React, { useEffect } from "react";
import { useNavigate } from "react-router-dom";
import SearchCard from "./SearchCard";
import WatchItLogo from "../WatchItLogo";

import "../../styles/searchresultscard.css";

function SearchResultsCard({ results }) {
  const navigate = useNavigate();

  // eslint-disable-next-line
  useEffect(() => {
    if (!results || results.length === 0) {
      const redirectTimer = setTimeout(() => {
        navigate("/");
        window.location.reload();
      }, 5000);

      return () => clearTimeout(redirectTimer);
    }
  }, [results, navigate]);

  return (
    <div>
      <WatchItLogo />
      {results && results.length > 0 ? (
        <div className="top-rated-movie-title">
          Search Results:
          <div
            className="search-results-card"
            data-testid="search-results-card"
          >
            {results.map((data) => (
              <div key={data.id} className="item">
                <SearchCard
                  title={data.title}
                  posterpath={data.poster_path}
                  movieId={data.id}
                  overview={data.overview}
                />
              </div>
            ))}
          </div>
        </div>
      ) : (
        <div className="no-results-container">
          <div className="no-results-message">
            No Results Found. Redirecting you back to the home page in 5
            seconds...
          </div>
        </div>
      )}
    </div>
  );
}

export default SearchResultsCard;
