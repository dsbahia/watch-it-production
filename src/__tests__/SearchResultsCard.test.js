import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import SearchResultsCard from "../components/Search/SearchResultsCard";

jest.mock("react-hot-toast", () => ({
  toast: {
    error: jest.fn(),
  },
}));

describe("SearchResultsCard", () => {
  const results = [
    {
      id: 1,
      title: "The Hunger Games",
      poster_path:
        "https://image.tmdb.org/t/p/original//4FAA18ZIja70d1Tu5hr5cj2q1sB.jpg",
    },
    {
      id: 2,
      title: "Star Wars",
      poster_path:
        "https://image.tmdb.org/t/p/original//6FfCtAuVAW8XJjZ7eWeLibRLWTw.jpg",
    },
  ];

  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <SearchResultsCard results={results} />
      </MemoryRouter>,
    );
    expect(asFragment).toMatchSnapshot();
  });

  it("renders SearchResultsCard with results", () => {
    render(
      <MemoryRouter>
        <SearchResultsCard results={results} />
      </MemoryRouter>,
    );

    results.forEach((data) => {
      expect(screen.getByText(data.title)).toBeInTheDocument();
      expect(
        screen.getByAltText(`${data.title} Movie poster`),
      ).toBeInTheDocument();
    });
  });

  it("renders nothing if no results", () => {
    render(
      <MemoryRouter>
        <SearchResultsCard results={[]} />
      </MemoryRouter>,
    );

    expect(screen.queryByTestId("search-results-card")).toBeNull();
  });
});
