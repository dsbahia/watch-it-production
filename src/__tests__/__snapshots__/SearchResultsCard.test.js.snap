// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsCard renders correctly 1`] = `[Function]`;
